fermipy/gtanalysis.py

```diff
@@ -4024,54 +4024,62 @@ def get_src_model(self, name, paramsonly=False, reoptimize=False,
             thesrc = self.like[name]
             src_dict['flux'] = self.like.flux(name, self.energies[0],
                                               self.energies[-1])
-            src_dict['flux100'] = self.like.flux(name, 100., 10 ** 5.5)
-            src_dict['flux1000'] = self.like.flux(name, 1000., 10 ** 5.5)
-            src_dict['flux10000'] = self.like.flux(name, 10000., 10 ** 5.5)
             src_dict['eflux'] = self.like.energyFlux(name,
                                                      self.energies[0],
                                                      self.energies[-1])
-            src_dict['eflux100'] = self.like.energyFlux(name, 100.,
-                                                        10 ** 5.5)
-            src_dict['eflux1000'] = self.like.energyFlux(name, 1000.,
-                                                         10 ** 5.5)
-            src_dict['eflux10000'] = self.like.energyFlux(name, 10000.,
-                                                          10 ** 5.5)
-            src_dict['dnde'] = self.like[name].spectrum()(
-                pyLike.dArg(src_dict['pivot_energy']))
-            src_dict['dnde100'] = self.like[name].spectrum()(
-                pyLike.dArg(100.))
-            src_dict['dnde1000'] = self.like[name].spectrum()(
-                pyLike.dArg(1000.))
-            src_dict['dnde10000'] = self.like[name].spectrum()(
-                pyLike.dArg(10000.))
+            emax = np.min([self.energies.max(), 10 ** 5.5])
+            for e in np.logspace(2.,4.,3):
+                k = "{0:n}".format(e)
+                if emax > e:
+                    src_dict['flux' + k] = self.like.flux(name, e, emax)
+                    src_dict['eflux' + k] = self.like.energyFlux(name, e,
+                                                        emax)
+                    src_dict['dnde' + k] = self.like[name].spectrum()(
+                            pyLike.dArg(e))
+                else:
+                    src_dict['flux' + k] = 0.
+                    src_dict['eflux' + k] = 0.
+                    src_dict['dnde' + k] = 0.
+
+            if emax > src_dict['pivot_energy']:
+                src_dict['dnde'] = self.like[name].spectrum()(
+                    pyLike.dArg(src_dict['pivot_energy']))
+            else:
```

Comment:
You might consider issuing a warning for cases like this one, so that it will be clear why the differential flux is zero. Indeed, despite the defaults, it would be better for the user to use a pivot energy within the energy range used for the analysis.